File: webapp/src/Websocket/types/Video.ts
export type VideoBrowserRequest = {
    minFileSize?: number;
    maxFileSize?: number;
    fromDate?: number;
    toDate?: number;
}

export type VideoBrowserResponse = {
    dirs: VideoDirectory[];
}

export type VideoDirectory = {
    dirname: string;
    files: VideoFile[];
}

export type VideoFile = {
    filename: string;
    size: number;
    timestampMillis: number;
}

File: src/main/kotlin/com/dehnes/smarthome/api/dtos/Video.kt
package com.dehnes.smarthome.api.dtos


data class VideoBrowserRequest(
    val minFileSize: Long?,
    val maxFileSize: Long?,
    val fromDate: Long?,
    val toDate: Long?
)

data class VideoBrowserResponse(
    val dirs: List<VideoDirectory>
)

data class VideoDirectory(
    val dirname: String,
    val files: List<VideoFile>
)

data class VideoFile(
    val filename: String,
    val size: Long,
    val timestampMillis: Long
)
